Mage.Sets/src/mage/cards/a/AnimalFriend.java

```diff
@@ -0,0 +1,95 @@
+package mage.cards.a;
+
+import java.util.UUID;
+
+import mage.abilities.Ability;
+import mage.abilities.common.AttacksTriggeredAbility;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.dynamicvalue.common.AuraAttachedCount;
+import mage.abilities.dynamicvalue.common.EquipmentAttachedCount;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.CreateTokenEffect;
+import mage.abilities.effects.common.continuous.GainAbilityAttachedEffect;
+import mage.constants.*;
+import mage.counters.CounterType;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.game.permanent.token.SquirrelToken;
+import mage.target.common.TargetCreaturePermanent;
+import mage.abilities.effects.common.AttachEffect;
+import mage.target.TargetPermanent;
+import mage.abilities.keyword.EnchantAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+
+/**
+ *
+ * @author grimreap124
+ */
+public final class AnimalFriend extends CardImpl {
+
+    public AnimalFriend(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{1}{G}");
+        
+        this.subtype.add(SubType.AURA);
+
+        // Enchant creature
+        TargetPermanent auraTarget = new TargetCreaturePermanent();
+        this.getSpellAbility().addTarget(auraTarget);
+        this.getSpellAbility().addEffect(new AttachEffect(Outcome.BoostCreature));
+        this.addAbility(new EnchantAbility(auraTarget));
+
+        // Enchanted creature has "Whenever this creature attacks, create a 1/1 green Squirrel creature token. Put a +1/+1 counter on that token for each Aura and Equipment attached to this creature other than Animal Friend."
+        Ability ability = new AttacksTriggeredAbility(new AnimalFriendEffect(), false);
+        this.addAbility(new SimpleStaticAbility(Zone.BATTLEFIELD, new GainAbilityAttachedEffect(ability, AttachmentType.AURA)));
+
+    }
+
+    private AnimalFriend(final AnimalFriend card) {
+        super(card);
+    }
+
+    @Override
+    public AnimalFriend copy() {
+        return new AnimalFriend(this);
+    }
+}
+
+class AnimalFriendEffect extends OneShotEffect {
+
+    AnimalFriendEffect() {
+        super(Outcome.PutCreatureInPlay);
+        staticText = "create a 1/1 green Squirrel creature token." +
+                " Put a +1/+1 counter on that token for each Aura and Equipment attached to this creature other than Animal Friend.";
```

Comment:
hmmm, this is interesting - the name ideally shouldn't be hardcoded, but it isn't "{this}", it's the attachment. Probably the most correct way to handle this is something along the lines of `GainAbilityWithAttachmentEffect`, but I think it's okay for now